src/graphql/object_type.cr

```diff
@@ -1,8 +1,12 @@
+require "./error"
 require "./language"
 require "./scalar_type"
 require "./internal/convert_value"
 
 module GraphQL::ObjectType
+  # :nodoc:
+  record JSONFragment, json : String, errors : Array(::GraphQL::Error)
```

Comment:
I think it's better to move this out of ObectType or it will be declared in every object class. And make sure to reference it as `::GraphQL::JSONFragment` in the macro code to avoid conflicts

Reply:
i'm pretty sure it isn't redeclared, as it's not in the included macro. At least while debugging i saw the type as GraphQL::ObjectType::JSONFragment (and not WhateverConcreteType::JSONFragment)

Reply:
Also see my comment above: no need to use `::` if I keep the methods outside of the macro, because there can't be conflicts when they're not declared in the including class